velox/row/UnsafeRow.h

```diff
@@ -401,17 +401,28 @@ class UnsafeRow {
 
     return std::string_view(buffer_ + offset, size);
   }
-
-  /**
-   * Write the data field as a continuous variable length data type at the end
-   * of the row.
-   * @param pos
-   * @param data a string_view over the variable length data.
-   */
-  void appendVariableLengthData(const std::string_view& data) {
-    std::memcpy(
-        fixedLengthData_ + variableLengthOffset_, data.begin(), data.size());
-  }
 };
 
+template <TypeKind kind>
+size_t serializedSizeInBytes() {
+  return sizeof(typename ScalarTraits<kind>::SerializedType);
+}
+
+template <>
+FOLLY_ALWAYS_INLINE size_t serializedSizeInBytes<TypeKind::VARCHAR>() {
+  VELOX_UNREACHABLE();
+}
+
+template <>
+FOLLY_ALWAYS_INLINE size_t serializedSizeInBytes<TypeKind::VARBINARY>() {
+  VELOX_UNREACHABLE();
+}
+
+/// Returns the number of bytes needed to serialized fixed-width type. Throws if
+/// 'type' is ot fixed-width.
+FOLLY_ALWAYS_INLINE size_t serializedSizeInBytes(const TypePtr& type) {
+  return VELOX_DYNAMIC_SCALAR_TYPE_DISPATCH(
```

Comment:
Hi, @mbasmanova.  I think it should use `VELOX_DYNAMIC_SCALAR_TYPE_DISPATCH_ALL`. otherwise decimal type will failed.

Reply:
Thanks, I got it. I just met a problem, because we support decimal types for UnsafeRow serialization in our oap/velox branch.

Reply:
> we support decimal types for UnsafeRow serialization in our oap/velox branch.

@zhejiangxiaomai Curious how do you serialize decimals? Have you modified the UnsafeRow serialization format somehow? Would you share some details? Hows the format you use compare with the one we use in Velox?

Reply:
Hi, @mbasmanova I'm not very familiar with this area. I think maybe this PR is answer. https://github.com/oap-project/velox/pull/196/files#diff-ab1c79f2ca46603dd123eb9fee40bbb3c20f6eef75837feeffa69eb23390587d
 cc:@jinchengchenghh @JkSelf please correct me if I'm wrong.

Reply:
This PR https://github.com/oap-project/velox/pull/196 serialize decimal in velox, spark decimal is a variable length value when it is int128_t.
The implementation breaks some Velox logic, it is not a completed final version.

Reply:
@jinchengchenghh Thank you for clarifying.

> spark decimal is a variable length value when it is int128_t

Got it. Is this how it is implemented in Spark as well?

CC: @majetideepak @karteekmurthys @aditi-pandit